Fix existing unit test, switch to pytest, add more

endolith opened the pull request:
Are you committed to the `unittest` framework?  I'm used to `pytest` and not sure what the equivalent of `@pytest.mark.parametrize` is.  

I also found [hypothesis](https://hypothesis.readthedocs.io/en/latest/) very useful in finding broken edge cases in my implementation, if you're ok with importing that.  https://github.com/endolith/elsim/blob/master/tests/test_star.py#L550

endolith committed:
Add Tennessee example

masiarek commented:
I prefer pytest as well..

mikefranze commented:
I'm not committed to any framework. You can change to pytest if you prefer it.

mikefranze commented:
My last PR also fixed the existing unit test and also changed the output format.

mikefranze commented on STAR_Test.py:
```diff
@@ -19,13 +19,26 @@ def test_original_example(self):
+                                     *15*[[0,      3,        5,          3]],
+                                     *17*[[0,      2,        4,          5]]])
+
+        self.assertEqual(STAR(ballots), 'Nashville')
```
Does this work? I think you need to pull out elected candidates from the results: self.assertEqual(results['elected'] , ['Nashville'])

endolith replied:
I had just rebased on top of the new output format without changing it. It should be fixed now

endolith committed:
Update tests for new output format

endolith committed:
Use pytest conventions

endolith committed:
Start adding star-core tests

mikefranze committed:
Bug in resolving ties when candidate indexes are strings

endolith committed:
Add star-core runner-up tie case

mikefranze committed:
Extend returns None, have to extend w first then return.

mikefranze committed:
Return tied object instead to clarify that tie occured

endolith committed:
Add other star-core tests

ArendPeter reviewed (approved):
Looks great! I'm also a pytest fan

endolith committed:
Add various tiebreaker test cases

I tried to enumerate every possible scenario involving ties in

https://github.com/endolith/elsim/blob/master/tests/test_star.py#L408

Some do not currently behave as expected and are commented out.
